python/test/light_TEST.py

```diff
@@ -0,0 +1,120 @@
+# You may obtain a copy of the License at
+
+#       http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+from gz.common import set_verbosity
+from gz_test_deps.sim import K_NULL_ENTITY, Light, TestFixture, World, world_entity
+from gz_test_deps.math import Angle, Color, Pose3d, Vector3d
+
+# This class test the Light class API created with pybind11. Since the setters methods
+# require other systems to work (eg. Rendering sensors), those methods will not be fully
+# tested, just verifying the method is being called.
+class TestLight(unittest.TestCase):
+    post_iterations = 0
+    iterations = 0
+    pre_iterations = 0
+
+    def test_light(self):
+        set_verbosity(4)
+
+        file_path = os.path.dirname(os.path.realpath(__file__))
+        fixture = TestFixture(os.path.join(file_path, 'light_test.sdf'))
+
+        def on_post_udpate_cb(_info, _ecm):
+            self.post_iterations += 1
+
+        def on_pre_udpate_cb(_info, _ecm):
+            self.pre_iterations += 1
+            world_e = world_entity(_ecm)
+            self.assertNotEqual(K_NULL_ENTITY, world_e)
+            w = World(world_e)
+            light_point = Light(w.light_by_name(_ecm, 'light_point_test'))
+            light_directional = Light(w.light_by_name(_ecm, 'light_directional_test'))
+            light_spot = Light(w.light_by_name(_ecm, 'light_spot_test'))
+            # Entity Test
+            self.assertNotEqual(K_NULL_ENTITY, light_point.entity())
+            self.assertNotEqual(K_NULL_ENTITY, light_directional.entity())
+            self.assertNotEqual(K_NULL_ENTITY, light_spot.entity())
+            # Valid Test
+            self.assertTrue(light_point.valid(_ecm))
+            self.assertTrue(light_directional.valid(_ecm))
+            self.assertTrue(light_spot.valid(_ecm))
+            # Name Test
+            self.assertEqual('light_point_test', light_point.name(_ecm))
+            self.assertEqual('light_directional_test', light_directional.name(_ecm))
+            self.assertEqual('light_spot_test', light_spot.name(_ecm))
+            # Pose Test
+            self.assertEqual(Pose3d(0, 2, 2, 0, 0, 0), light_point.pose(_ecm))
+            light_point.set_pose(_ecm, Pose3d(4, 2, 2, 0, 0, 0))
+            # Type Test
+            self.assertEqual('point', light_point.type(_ecm))
+            self.assertEqual('directional', light_directional.type(_ecm))
+            self.assertEqual('spot', light_spot.type(_ecm))
+            ## Diffuse Color Test
+            self.assertEqual(Color(1, 0, 0, 1), light_point.diffuse_color(_ecm))
+            light_point.set_diffuse_color(_ecm, Color(1, 1, 0, 1))
```

Comment:
Is there supposed to be a test here affter seting the diffuse color to check that it worked? Same for specular, cast shadows, etc.

Reply:
I added them, however, they are not working for some reason, I pushed and example for you to take a look.

Reply:
@iche033 It looks like the getter (`Light::DiffuseColor`) returns the value of the `sdf::Light` which was in the original SDF file, not the current state of the light in the scene. Is that the intended behavior?

Reply:
I've created an issue https://github.com/gazebosim/gz-sim/issues/2095. For now, let's just skip the tests for the `set*` functions.

Reply:
It turns out the getters do work, but a rendering system, such as the Sensor system (and a camera) is required. I've removed the checks that require this in https://github.com/gazebosim/gz-sim/pull/2043/commits/209e5562d55cc21186bf11c9d02e15dc216841a7